application/slack_messenger/slack_messenger.py

```diff
@@ -0,0 +1,176 @@
+                        f"{str(trigger['Threshold'])} for {str(trigger['EvaluationPeriods'])} period(s) "
+                        f" of {str(trigger['Period'])} seconds.",
+                        "short": False,
+                    },
+                    {
+                        "title": "",
+                        "value": f"<{generate_cloudwatch_url(region, log_groups, filters)}|View Logs>",
+                    },
+                ],
+                "ts": timestamp,
+            }
+        ],
+    }
+    return slack_message
+
+
+def get_report_key(metric_name):
+    if metric_name == "InvalidPostcode":
+        return INVALID_POSTCODE_REPORT_ID
+    elif metric_name == "InvalidOpenTimes":
+        return INVALID_OPEN_TIMES_REPORT_ID
+    return ""
+
+
+def send_msg_slack(message):
+    url = environ["SLACK_WEBHOOK_URL"]
+    channel = environ["SLACK_ALERT_CHANNEL"]
+    headers: Dict[str, str] = {"Content-Type": "application/json", "Accept": "application/json"}
+
+    message["channel"] = channel
+    message["icon_emoji"] = ""
+
+    resp = post(
+        url=url,
+        headers=headers,
+        json=message,
+    )
+    logger.info(
+        "Message sent to slack",
+        extra={"slack_message": message, "status_code": resp.status_code, "response": resp.text},
+    )
+
+
+def generate_cloudwatch_url(region: str, log_groups: list, filters: dict, limit: int = 100):
+    def escape(s):
+        for c in s:
+            if c.isalpha() or c.isdigit() or c in ["-", "."]:
+                continue
+            c_hex = "*{0:02x}".format(ord(c))
+            s = s.replace(c, c_hex)
+        return s
+
+    def generate_log_insights_url(params):
+        S1 = "$257E"
+        S2 = "$2528"
+        S3 = "$2527"
+        S4 = "$2529"
+
+        res = f"{S1}{S2}"
+        for k in params:
+            value = params[k]
+            if isinstance(value, str):
+                value = escape(value)
+            elif isinstance(value, list):
+                for i in range(len(value)):
+                    value[i] = escape(value[i])
+            prefix = S1 if list(params.items())[0][0] != k else ""
+            suffix = f"{S1}{S3}"
+            if isinstance(value, list):
+                value = "".join([f"{S1}{S3}{n}" for n in value])
+                suffix = f"{S1}{S2}"
+            elif isinstance(value, int) or isinstance(value, bool):
+                value = str(value).lower()
+                suffix = S1
+            res += f"{prefix}{k}{suffix}{value}"
+        res += f"{S4}{S4}"
+        QUERY = f"logsV2:logs-insights$3Ftab$3Dlogs$26queryDetail$3D{res}"
+        return f"https://{region}.console.aws.amazon.com/cloudwatch/home?region={region}#{QUERY}"
+
+    query = "\n".join([f'| filter {k}="{v}"' for (k, v) in filters.items()])
+    fields = "fields @timestamp,correlation_id,ods_code,level,message_received,function_name, message"
+    params = {
+        "end": 0,
+        "start": -60 * 60 * 1,  # - 1hr
```

Comment:
might be better to use current time - 1 hour, this way if someone doesnt get to the alert for an hour, the logs will still contain data that triggerd rather than data an hour later?

Reply:
The current implementation is current time - hr.  But this can be changed in Log Insights once they click on the link though